components/style/restyle_hints.rs

```diff
@@ -161,7 +161,7 @@ impl<'a, E> Element for ElementWrapper<'a, E>
                 match attr.namespace {
                     NamespaceConstraint::Specific(ref ns) => self.snapshot.get_attr(ns, local_name),
                     NamespaceConstraint::Any => self.snapshot.get_attr_ignore_ns(local_name),
-                }.map_or(false, |v| test(v))
+                }.map_or(false, |v| test(&v.to_string()))
```

Comment:
What does the to_string implementation in string_cache look like? Does it copy? If so, I'm not sure if it's acceptable to use here.